apps/base-docs/docs/pages/identity/smart-wallet/guides/linked-accounts/creating-sub-accounts.mdx

```diff
@@ -0,0 +1,164 @@
+---
+title: "Creating Sub Accounts"
+description: "Learn how to create and manage sub accounts with Coinbase Smart Wallet"
+---
+
+# Creating Sub Accounts
+
+After setting up your project with Coinbase Smart Wallet integration, the next step is implementing sub account creation. Sub accounts allow your application to create wallet accounts that are linked to the user's main Coinbase Smart Wallet.
+
+## Configuring the SDK for Sub Accounts
+
+First, we need to update our SDK configuration to enable sub account creation. The SDK provides browser key signers out of the box, which handle the creation and caching of signers for you.
+
+Update your CoinbaseWalletContext.tsx file to include sub account configuration:
+
+```tsx
+// app/context/CoinbaseWalletContext.tsx
+import { 
+  createCoinbaseWalletSDK, 
+  getCryptoKeyAccount, 
+  ProviderInterface 
+} from '@coinbase/wallet-sdk';
+
+// Update the SDK initialization in useEffect
+useEffect(() => {
+  const coinbaseWalletSDK = createCoinbaseWalletSDK({
+    appName: 'Coinbase Wallet demo',
+    appChainIds: [baseSepolia.id],
+    preference: {
+      options: "smartWalletOnly",
+      keysUrl: 'https://keys-dev.coinbase.com/connect',
+    },
+    subaccount: {
+      getSigner: getCryptoKeyAccount,
+    }
+  });
+
+  setProvider(coinbaseWalletSDK.getProvider());
+}, []);
+```
+
+## Adding Sub Account State Management
+
+Next, we'll add state management for sub accounts to our context:
+
+```tsx
+// app/context/CoinbaseWalletContext.tsx
+
+interface CoinbaseWalletContextType {
+  // ... existing interface properties
+  subAccount: Address | null;
+  createSubAccount: () => Promise<Address>;
+}
+
+export function CoinbaseWalletProvider({ children }: { children: ReactNode }) {
+  // ... existing state
+  const [subAccount, setSubAccount] = useState<Address | null>(null);
+
+  const createSubAccount = useCallback(async () => {
+    if (!provider || !address) {
+      throw new Error('Address or provider not found');
+    };
+    
+    const signer = await getCryptoKeyAccount();
+
+    const walletConnectResponse = await provider.request({
+      method: 'wallet_connect',
+      params: [{
+        version: '1',
+        capabilities: {
+          addAddress: {
+            chainId: baseSepolia.id,
+            createAccount: {
+              signer: signer.account?.publicKey,
+            },
+            address
+          },
+        },
+      }],
+    });
+
+    const subAccount = walletConnectResponse?.accounts[0].capabilities?.addAddress?.address;
```

Comment:
this feels pretty tough, this is how we discover if an account is a sub account?

Reply:
yeah the other option is use a separate RPC